disperser/dataapi/subgraph_client.go

```diff
@@ -144,6 +219,86 @@ func (sc *subgraphClient) QueryIndexedDeregisteredOperatorsInTheLast14Days(ctx c
 	}, nil
 }
 
+func getOperatorsWithRegisteredDeregisteredIntervalEvents(
+	ctx context.Context,
+	registeredOperators []*subgraph.Operator,
+	deregisteredOperators []*subgraph.Operator,
+) (chan OperatorEvents, error) {
+	sort.SliceStable(registeredOperators, func(i, j int) bool {
+		return registeredOperators[i].BlockTimestamp < registeredOperators[j].BlockTimestamp
+	})
+
+	sort.SliceStable(deregisteredOperators, func(i, j int) bool {
+		return deregisteredOperators[i].BlockTimestamp < deregisteredOperators[j].BlockTimestamp
+	})
+
+	// First position is for registered events and second position is for deregistered events
+	operators := make(map[string][][]uint64, 0)
+	for i := range registeredOperators {
+		operator := registeredOperators[i]
+		operatorId := string(operator.OperatorId)
+
+		if _, ok := operators[operatorId]; !ok {
+			operators[operatorId] = make([][]uint64, 2)
+		}
+		timestamp, err := strconv.ParseUint(string(operator.BlockTimestamp), 10, 64)
+		if err != nil {
+			return nil, err
+		}
+		operators[operatorId][0] = append(operators[operatorId][0], timestamp)
+	}
+
+	for i := range deregisteredOperators {
+		operator := deregisteredOperators[i]
+		operatorId := string(operator.OperatorId)
+
+		timestamp, err := strconv.ParseUint(string(operator.BlockTimestamp), 10, 64)
+		if err != nil || timestamp == 0 {
+			return nil, err
+		}
+		if _, ok := operators[operatorId]; ok {
+			operators[operatorId][1] = append(operators[operatorId][1], timestamp)
+		}
+	}
+
+	operatorEventsChan := make(chan OperatorEvents, 1)
+	go func() {
+		defer close(operatorEventsChan)
+		pool := workerpool.New(10)
+		for o, e := range operators {
+			operatorId, events := o, e
+			pool.Submit(func() {
+				var lastDeregistered uint64
+				for i := 0; i < len(events[0]); i++ {
+					var (
+						// we start with the assumption that the operator is still registered
+						// a 0 value means that the operator is still registered
+						deregistered uint64
+						registered   = events[0][i]
+					)
+
+					// if there is a deregistered event that macthes in same position of registered event, we use it
+					if i < len(events[1]) {
+						deregistered = events[1][i]
+					}
+
+					// if there is no deregistered event that matches in same position of registered event, we use the last one
+					if i == 0 || registered > lastDeregistered {
```

Comment:
what if they deregister and register in the same block?

Reply:
That should not have effect on operator's responsibility: if this block is taken as reference block, then the operator is considered as in the network for signing.